barretenberg/cpp/src/barretenberg/vm2/simulation/standalone/pure_bytecode_manager.cpp

```diff
@@ -61,16 +61,25 @@ BytecodeId PureTxBytecodeManager::get_bytecode(const AztecAddress& address)
 
     retrieved_class_ids.insert(current_class_id);
 
-    // Contract class retrieval and class ID validation
+    // Contract class retrieval
     std::optional<ContractClass> maybe_klass = contract_db.get_contract_class(current_class_id);
-    // Note: we don't need to silo and check the class id because the deployer contract guarrantees
+    // Note: we don't need to silo and check the class id because the deployer contract guarantees
     // that if a contract instance exists, the class has been registered.
     assert(maybe_klass.has_value());
     auto& klass = maybe_klass.value();
     debug("Bytecode for ", address, " successfully retrieved!");
 
     // Bytecode hashing and decomposition, deduplicated by bytecode_id (commitment)
-    BytecodeId bytecode_id = klass.public_bytecode_commitment;
+    // TODO(dbanks12): in TS, the PublicContractsDB will hash the bytecode if it has never been hashed there before.
+    // After that, it caches it. It should only happen once per contract class, but when we are making a callback
+    // to the TS cache to hash the bytecode there, it might be unnecessarily slow, in which case we could do the same
+    // hashing and caching here in C++ and avoid callbacks to TS. Note that separating out the get_bytecode_commitment
+    // DB call was NOT a regression in this regard as the same TS hashing would have happened when our
+    // get_contract_class function made a TS callback to get_bytecode_commitment.
+    std::optional<FF> maybe_bytecode_commitment = contract_db.get_bytecode_commitment(current_class_id);
+    // If we reach this point, class ID and instance both exist which means bytecode commitment must exist.
+    assert(maybe_bytecode_commitment.has_value());
+    BytecodeId bytecode_id = maybe_bytecode_commitment.value();
```

Comment:
You know what... what if in the pure version we deduplicate by class id? Then we never need to ask about the commitment. I'm not sure about the cost of the TS roundtrip. Do you have any measurements in general of the overhead of using the callback? Even just some rough numbers of how long it takes to simulate app enqueued calls, comparing with the numbers I posted some time ago, should give us an idea.

I'm not super against the current solution, but I do dislike that we cannot assume that the DB has the commitment (not because of this PR). I'm pretty sure that might make some tests fail when we want to stop using the TS callback. At some point we might want to either be sure you can, or push the computation to the DB layer.